ast/src/main/java/com/graphicsfuzz/common/tool/PrettyPrinterVisitor.java

```diff
@@ -172,10 +204,43 @@ public void visitDeclarationStmt(DeclarationStmt declarationStmt) {
 
   @Override
   public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration) {
+
+    // If uniform variables are being declared and we have a uniform supplier that provides known
+    // values for one or more of the uniforms, we emit details of those known values in comments,
+    // one line per uniform.
+    if (variablesDeclaration.getBaseType().hasQualifier(TypeQualifier.UNIFORM)
+            && uniformValues.isPresent()) {
+      // Find the variable declarations for which we have known uniform values.
+      final List<VariableDeclInfo> knownUniforms =
+          variablesDeclaration.getDeclInfos().stream()
+              .filter(uniform -> uniformValues.get()
+                  .getValues(uniform.getName()).isPresent()).collect(Collectors.toList());
+
+      // Emit a comment per known uniform.
+      for (VariableDeclInfo knownUniform : knownUniforms) {
+        final Optional<List<String>> values = uniformValues.get().getValues(knownUniform.getName());
+        values.ifPresent(item -> {
+          out.append("// Contents of ")
+              .append(knownUniform.getName())
+              .append(": ");
+          if (item.size() > 1) {
+            out.append("[");
+          }
+          out.append(item.stream().map(Object::toString)
+              .reduce((item1, item2) -> item1 + ", " + item2).orElse(""));
+          if (item.size() > 1) {
+            out.append("]");
+          }
+        });
+        out.append("\n");
+      }
+    }
+
     final Type baseType = variablesDeclaration.getBaseType();
     visit(baseType);
     out.append(" ");
     boolean first = true;
+
```

Comment:
Avoid adding unnecessary whitespace in a PR (unless the PR is a "code cleanup" PR).  (Apologies if this came from my patch!)